File: Data Structures & Algorithms/search-2d-matrix/submission-1.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int low =0 ,high = matrix.length-1;
        
        while(low<=high){
            int mid = low+(high-low)/2;
            if(matrix[mid][matrix[0].length-1]<target){
                low= mid+1;
            }else if(matrix[mid][matrix[0].length-1]>target){
                high = mid-1;
            }
            int low2 =0 ,high2=matrix[0].length-1;
          while (low2 <= high2) {
    int m = low2 + (high2 - low2) / 2;
    if (target < matrix[mid][m]) {
        high2 = m - 1;
    } else if (target > matrix[mid][m]) {
        low2 = m + 1;
    } else {
        return true;
    }
}

            
        }
        return false ;
    }
}

File: Data Structures & Algorithms/arranging-coins/submission-1.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        l,r=0,n
        while l<=r:
            mid = (l+r)//2
            sumUntilMid = (mid*(mid+1))//2
            if sumUntilMid<=n:
                result = mid
                l=mid+1

            else:
                r=mid-1
        return result